src/redux/sagas/survey.saga.js

```diff
@@ -0,0 +1,31 @@
+import { put, takeLatest } from 'redux-saga/effects';
+import axios from 'axios';
+
+// worker saga: will be fired on "CREATE_PREFERENCE"
+function* createPreference(action) {
+    console.log('in createPreference saga');
+    // check payload
+    console.log('preference saga payload', action.payload);
+
+    try{
+        yield axios.post('/api/survey', action.payload);
```

Comment:
imo this should be an axios.put(XXX) because we are not creating new survey results, just editing existing ones that should be created when the user registers an account.

Reply:
It is a POST because we didn't populate the table initially. That's what I was suggesting. There can't be a PUT request for a data that's yet to exist.

Reply:
But I can make it a PUT by auto populating the preference table on the registering of a new user. Should I do that?

Reply:
i would say yes. because if you POST every time they change the values you're going to create a ton of bugs because there will be multiple rows in the table per user with repeated value IDs

Reply:
specifically: yes make a POST upon user registration, and change the stuff you already did to a PUT

Reply:
Yes... I realized this so I stopped... I think I put it on the Trello board that it needs to be restructured. And that starts from the registration component. But yeah, it's going to be straightforward. I know how it's going to work. And for fun, I could try await async. 😃

Reply:
that might be good! here's where i added a little async..await in my solo project: [Link Here](https://github.com/jamespetran/Satellite-Tracker/blob/e952397210bcb7f1e228bf1ed5141d1b2c8fe23b/server/routes/user.router.js#L106)